components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -255,16 +256,18 @@ private String getAppID(String appName, String tenantDomain) throws IdentityOAut
     }
 
     /**
-     * Validate grant type of access token.
+     * Change user type for tokens switched with client credentials grant as APPLICATION.
      *
+     * @param tokReqMsgCtx  token request message context
      * @param accessTokenDO access token to be switched
      */
-    protected void validateGrantType(AccessTokenDO accessTokenDO) throws IdentityOAuth2Exception {
-
-        if (OAuthConstants.GrantTypes.CLIENT_CREDENTIALS.equals(accessTokenDO.getGrantType())) {
-            LOG.debug("Access token validation failed.");
+    private void changeUserTypeForCCGrant(OAuthTokenReqMessageContext tokReqMsgCtx, AccessTokenDO accessTokenDO) {
 
-            throw new IdentityOAuth2Exception("Invalid grant received.");
+        if (OAuthConstants.GrantTypes.CLIENT_CREDENTIALS.equals(accessTokenDO.getGrantType()) ||
+                (OAuthConstants.GrantTypes.ORGANIZATION_SWITCH.equals(accessTokenDO.getGrantType()) &&
+                        OAuthConstants.UserType.APPLICATION.equals(tokReqMsgCtx.getProperty(
+                                tokReqMsgCtx.getProperty(OAuthConstants.UserType.USER_TYPE).toString())))) {
```

Comment:
the logic here is unclear

Reply:
The logic should be corrected.
There's no need to check USER_TYPE if the grant type is ORGANIZATION_SWITCH.
Addressed with d06e312ee688b79a3c12553b27b9672a37863ad5